azurerm/internal/services/sentinel/sentinel_automation_rule_resource_test.go

```diff
@@ -0,0 +1,250 @@
+  order                      = 2
+  condition {
+    property = "IncidentTitle"
+    operator = "Contains"
+    values   = ["a", "b"]
+  }
+
+  condition {
+    property = "IncidentTitle"
+    operator = "Contains"
+    values   = ["c", "d"]
+  }
+
+  action_incident {
+    order                  = 1
+    status                 = "Closed"
+    classification         = "BenignPositive_SuspiciousButExpected"
+    classification_comment = "whatever reason"
+  }
+
+  action_incident {
+    order  = 3
+    labels = ["foo", "bar"]
+  }
+
+  action_incident {
+    order    = 2
+    severity = "High"
+  }
+
+  action_incident {
+    order    = 4
+    owner_id = data.azurerm_client_config.current.object_id
+  }
+}
+`, template, r.uuid, data.RandomInteger)
+}
+
+func (r SentinelAutomationRuleResource) requiresImport(data acceptance.TestData) string {
+	template := r.basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_sentinel_automation_rule" "import" {
+  name                       = azurerm_sentinel_automation_rule.test.name
+  log_analytics_workspace_id = azurerm_sentinel_automation_rule.test.log_analytics_workspace_id
+  display_name               = azurerm_sentinel_automation_rule.test.display_name
+  order                      = azurerm_sentinel_automation_rule.test.order
+  action_incident {
+    order = 1
+  }
+}
+`, template)
+}
+
+func (r SentinelAutomationRuleResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-sentinel-%d"
+  location = %q
+}
+
+resource "azurerm_log_analytics_workspace" "test" {
+  name                = "acctest-workspace-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku                 = "pergb2018"
+}
+
+resource "azurerm_log_analytics_solution" "sentinel" {
+  solution_name         = "SecurityInsights"
+  location              = azurerm_resource_group.test.location
+  resource_group_name   = azurerm_resource_group.test.name
+  workspace_resource_id = azurerm_log_analytics_workspace.test.id
+  workspace_name        = azurerm_log_analytics_workspace.test.name
+
+  plan {
+    publisher = "Microsoft"
+    product   = "OMSGallery/SecurityInsights"
+  }
+}
+
+resource "azurerm_sentinel_alert_rule_scheduled" "test" {
+  name                       = "acctest-SentinelAlertRule-Sche-%d"
+  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
+  display_name               = "Some Rule"
+  severity                   = "High"
+  query                      = <<QUERY
+AzureActivity |
+  where OperationName == "Create or Update Virtual Machine" or OperationName =="Create Deployment" |
+  where ActivityStatus == "Succeeded" |
+  make-series dcount(ResourceId) default=0 on EventSubmissionTimestamp in range(ago(7d), now(), 1d) by Caller
+QUERY
+}
```

Comment:
```suggestion
func (r SentinelAutomationRuleResource) complete(data acceptance.TestData) string {
	template := r.template(data)
	return fmt.Sprintf(`
%s

data "azurerm_client_config" "current" {}

resource "azurerm_sentinel_automation_rule" "test" {
  depends_on = [azurerm_template_deployment.testlogicapp]
  name                       = "%s"
  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
  display_name               = "acctest-SentinelAutoRule-%d-update"
  order                      = 2
  condition {
    property = "IncidentTitle"
    operator = "Contains"
    values   = ["a", "b"]
  }

  condition {
    property = "IncidentTitle"
    operator = "Contains"
    values   = ["c", "d"]
  }

  action_incident {
    order                  = 1
    status                 = "Closed"
    classification         = "BenignPositive_SuspiciousButExpected"
    classification_comment = "whatever reason"
  }

  action_incident {
    order  = 3
    labels = ["foo", "bar"]
  }

  action_incident {
    order    = 2
    severity = "High"
  }

  action_incident {
    order    = 4
    owner_id = data.azurerm_client_config.current.object_id
  }

  action_playbook {
    order    = 5
    logic_app_id = "/subscriptions/${data.azurerm_subscription.current.subscription_id}/resourceGroups/${azurerm_resource_group.test.name}/providers/Microsoft.Logic/workflows/Test"
  }

}
`, template, r.uuid, data.RandomInteger)
}

func (r SentinelAutomationRuleResource) requiresImport(data acceptance.TestData) string {
	template := r.basic(data)
	return fmt.Sprintf(`
%s

resource "azurerm_sentinel_automation_rule" "import" {
  name                       = azurerm_sentinel_automation_rule.test.name
  log_analytics_workspace_id = azurerm_sentinel_automation_rule.test.log_analytics_workspace_id
  display_name               = azurerm_sentinel_automation_rule.test.display_name
  order                      = azurerm_sentinel_automation_rule.test.order
  action_incident {
    order = 1
  }
}
`, template)
}

func (r SentinelAutomationRuleResource) template(data acceptance.TestData) string {
	return fmt.Sprintf(`
provider "azurerm" {
  features {}
}

data "azurerm_subscription" "current" {
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-sentinel-%d"
  location = %q
}

resource "azurerm_log_analytics_workspace" "test" {
  name                = "acctest-workspace-%d"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  sku                 = "pergb2018"
}

resource "azurerm_log_analytics_solution" "sentinel" {
  solution_name         = "SecurityInsights"
  location              = azurerm_resource_group.test.location
  resource_group_name   = azurerm_resource_group.test.name
  workspace_resource_id = azurerm_log_analytics_workspace.test.id
  workspace_name        = azurerm_log_analytics_workspace.test.name

  plan {
    publisher = "Microsoft"
    product   = "OMSGallery/SecurityInsights"
  }
}

resource "azurerm_sentinel_alert_rule_scheduled" "test" {
  name                       = "acctest-SentinelAlertRule-Sche-%d"
  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
  display_name               = "Some Rule"
  severity                   = "High"
  query                      = <<QUERY
AzureActivity |
  where OperationName == "Create or Update Virtual Machine" or OperationName =="Create Deployment" |
  where ActivityStatus == "Succeeded" |
  make-series dcount(ResourceId) default=0 on EventSubmissionTimestamp in range(ago(7d), now(), 1d) by Caller
QUERY
}

resource "azurerm_template_deployment" "testlogicapp" {
	depends_on = [azurerm_template_deployment.testconnection]
	name = "testlogicapp"
	resource_group_name = "${azurerm_resource_group.test.name}"
	deployment_mode = "Incremental"
	template_body = <<TEMPLATE
  {
	  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
	  "contentVersion": "1.0.0.0",
	  "parameters": {
		  "workflows_Test_name": {
			  "defaultValue": "Test",
			  "type": "String"
		  },
		  "connections_azuresentinel_externalid": {
			  "defaultValue": "/subscriptions/${data.azurerm_subscription.current.subscription_id}/resourceGroups/${azurerm_resource_group.test.name}/providers/Microsoft.Web/connections/azuresentinel",
			  "type": "String"
		  }
	  },
	  "variables": {},
	  "resources": [
		  {
			  "type": "Microsoft.Logic/workflows",
			  "apiVersion": "2017-07-01",
			  "name": "[parameters('workflows_Test_name')]",
			  "location": "${azurerm_resource_group.test.location}",
			  "properties": {
				  "state": "Enabled",
				  "definition": {
					  "$schema": "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
					  "contentVersion": "1.0.0.0",
					  "parameters": {
						  "$connections": {
							  "defaultValue": {},
							  "type": "Object"
						  }
					  },
					  "triggers": {
						  "When_Azure_Sentinel_incident_creation_rule_was_triggered": {
							  "type": "ApiConnectionWebhook",
							  "inputs": {
								  "body": {
									  "callback_url": "@{listCallbackUrl()}"
								  },
								  "host": {
									  "connection": {
										  "name": "@parameters('$connections')['azuresentinel']['connectionId']"
									  }
								  },
								  "path": "/incident-creation"
							  }
						  }
					  },
					  "actions": {
						  "Add_comment_to_incident_(V3)": {
							  "runAfter": {},
							  "type": "ApiConnection",
							  "inputs": {
								  "body": {
									  "incidentArmId": "@triggerBody()?['object']?['id']",
									  "message": "<p>HELLO WORLD</p>"
								  },
								  "host": {
									  "connection": {
										  "name": "@parameters('$connections')['azuresentinel']['connectionId']"
									  }
								  },
								  "method": "post",
								  "path": "/Incidents/Comment"
							  }
						  }
					  },
					  "outputs": {}
				  },
				  "parameters": {
					  "$connections": {
						  "value": {
							  "azuresentinel": {
								  "connectionId": "[parameters('connections_azuresentinel_externalid')]",
								  "connectionName": "azuresentinel",
								  "id": "/subscriptions/${data.azurerm_subscription.current.subscription_id}/providers/Microsoft.Web/locations/${azurerm_resource_group.test.location}/managedApis/azuresentinel"
							  }
						  }
					  }
				  }
			  }
		  }
	  ]
  }
  TEMPLATE
	
  }

  resource "azurerm_template_deployment" "testconnection" {
	name = "testconnection"
	resource_group_name = "${azurerm_resource_group.test.name}"
	deployment_mode = "Incremental"
	template_body = <<TEMPLATE
	{
		"$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
		"contentVersion": "1.0.0.0",
		"parameters": {
			"connections_azuresentinel_name": {
				"defaultValue": "azuresentinel",
				"type": "String"
			}
		},
		"variables": {},
		"resources": [
			{
				"type": "Microsoft.Web/connections",
				"apiVersion": "2016-06-01",
				"name": "[parameters('connections_azuresentinel_name')]",
				"location": "${azurerm_resource_group.test.location}",
				"kind": "V1",
				"properties": {
					"displayName": "test",
					"customParameterValues": {},
					"api": {
						"id": "[concat('/subscriptions/${data.azurerm_subscription.current.subscription_id}/providers/Microsoft.Web/locations/${azurerm_resource_group.test.location}/managedApis/', parameters('connections_azuresentinel_name'))]"
					}
				}
			}
		]
	}
  TEMPLATE
	
  }
```

These tests can pass with given configuration as a logic app can still be attached with a invalid API connection, as long as the api connection exists.
![image](https://user-images.githubusercontent.com/29356754/116629701-b63ba100-a949-11eb-9d9f-6ad61080e5d1.png)

We should probably warn users somewhere that role assignments such as Azure Sentinel Automation Contributor need to be made somewhere, unless we specify in env vars we require the ID Of Azure Security Insights principle to make our own role assignment. I.e
```
fmt.Printf("=== WARN: You must assign the Azure Sentinel Automation Contributor role on the Test subscription to the 'Azure Security Insights' principle. This ObjectID Is a unique GUID per tenant. Expect this check to fail otherwise as Sentinel needs permission on Logic App resource groups for run in addition to user permissions\n")
```

Bit of a hacky workaround the lack of support that uses template deployments so some manual dependency specifications in here

```
=== RUN   TestAccSentinelAutomationRule_basic
=== WARN: You must assign the Azure Sentinel Automation Contributor role on the Test subscription to the 'Azure Security Insights' principle. This ObjectID Is a unique GUID per tenant. Expect this check to fail otherwise as Sentinel needs permission on Logic App resource groups for run in addition to user permissions
=== PAUSE TestAccSentinelAutomationRule_basic
=== RUN   TestAccSentinelAutomationRule_complete
=== PAUSE TestAccSentinelAutomationRule_complete
=== RUN   TestAccSentinelAutomationRule_update
=== PAUSE TestAccSentinelAutomationRule_update
=== RUN   TestAccSentinelAutomationRule_requiresImport
=== PAUSE TestAccSentinelAutomationRule_requiresImport
=== CONT  TestAccSentinelAutomationRule_basic
=== CONT  TestAccSentinelAutomationRule_requiresImport
=== CONT  TestAccSentinelAutomationRule_update
=== CONT  TestAccSentinelAutomationRule_complete
--- PASS: TestAccSentinelAutomationRule_basic (422.91s)
--- PASS: TestAccSentinelAutomationRule_complete (426.75s)
--- PASS: TestAccSentinelAutomationRule_requiresImport (436.83s)
--- PASS: TestAccSentinelAutomationRule_update (634.81s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel    634.918s
```